### ISSUE TITLE
An error occurred while Style/IfUnlessModifier cop was inspecting <file>

### ISSUE THREAD

**Issue by erwanst** [rubocop/rubocop#11282]
## Expected behavior

The `Style/IfUnlessModifier` doesn't produce an error when inspecting valid Ruby code. 

## Actual behavior

The cop raised the following exception when inspecting a valid conditional:

```
undefined method `value_omission?' for nil:NilClass

           if_body.last_argument&.hash_type? && if_body.last_argument.pairs.last.value_omission?
```

The code in question is:

```ruby
# frozen_string_literal: true

def foo
  if bar['baz']
    bar['baz'] = {}
  end
end
```

## Steps to reproduce the problem

Given the following files:

_Gemfile_

```ruby
# frozen_string_literal: true

source 'https://rubygems.org'

gem 'rubocop', '1.40.0'
```

_Gemfile.lock_

```
GEM
  remote: https://rubygems.org/
  specs:
    ast (2.4.2)
    json (2.6.3)
    parallel (1.22.1)
    parser (3.1.3.0)
      ast (~> 2.4.1)
    rainbow (3.1.1)
    regexp_parser (2.6.1)
    rexml (3.2.5)
    rubocop (1.40.0)
      json (~> 2.3)
      parallel (~> 1.10)
      parser (>= 3.1.2.1)
      rainbow (>= 2.2.2, < 4.0)
      regexp_parser (>= 1.8, < 3.0)
      rexml (>= 3.2.5, < 4.0)
      rubocop-ast (>= 1.23.0, < 2.0)
      ruby-progressbar (~> 1.7)
      unicode-display_width (>= 1.4.0, < 3.0)
    rubocop-ast (1.24.0)
      parser (>= 3.1.1.0)
    ruby-progressbar (1.11.0)
    unicode-display_width (2.3.0)

PLATFORMS
  x86_64-linux

DEPENDENCIES
  rubocop (= 1.40.0)

BUNDLED WITH
   2.3.26
```

_ok.rb_

```ruby
# frozen_string_literal: true

def foo
  if bar['baz']
    bar['baz'] = baz
  end
end
```

_ko.rb_

```ruby
# frozen_string_literal: true

def foo
  if bar['baz']
    bar['baz'] = {}
  end
end
```

_report.sh_

```bash
#!/usr/bin/env bash

set -x

bundle exec rubocop -d --only Style/IfUnlessModifier -- ok.rb
bundle exec rubocop -d --only Style/IfUnlessModifier -- ko.rb
```

Executing `report.sh` will produce the following log, where inspecting the `ok.rb` file doesn't fail but inspecting `ko.rb` does.

_report.log_

```
+ bundle exec rubocop -d --only Style/IfUnlessModifier -- ok.rb
For ~/ifunless: Default configuration from ~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning ~/ifunless/ok.rb
Loading cache from ~/.cache/rubocop_cache/f3efb5900a1b92b56bd0cd85eeaddbff3cd04234/075f5f40810684df981dea75a2ec46dc2ce75e1c/f1d62ebc72aeb2b9fef57abb03fd2069da3c4e84
C

Offenses:

ok.rb:4:3: C: [Correctable] Style/IfUnlessModifier: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
  if bar['baz']
  ^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.0713412100003552 seconds
+ bundle exec rubocop -d --only Style/IfUnlessModifier -- ko.rb
An error occurred while Style/IfUnlessModifier cop was inspecting ~/ifunless/ko.rb:4:2.

1 error occurred:
An error occurred while Style/IfUnlessModifier cop was inspecting ~/ifunless/ko.rb:4:2.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.40.0 (using Parser 3.1.3.0, rubocop-ast 1.24.0, running on ruby 3.1.3) [x86_64-linux]
For ~/ifunless: Default configuration from ~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning ~/ifunless/ko.rb
undefined method `value_omission?' for nil:NilClass

           if_body.last_argument&.hash_type? && if_body.last_argument.pairs.last.value_omission?
                                                                                ^^^^^^^^^^^^^^^^
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/mixin/statement_modifier.rb:59:in `if_body_source'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/mixin/statement_modifier.rb:51:in `to_modifier_form'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/mixin/statement_modifier.rb:46:in `length_in_modifier_form'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/mixin/statement_modifier.rb:40:in `modifier_fits_on_single_line?'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/mixin/statement_modifier.rb:16:in `single_line_as_modifier?'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/style/if_unless_modifier.rb:54:in `on_if'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:100:in `public_send'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:100:in `block (2 levels) in trigger_responding_cops'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:160:in `with_cop_error_handling'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:99:in `block in trigger_responding_cops'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:98:in `each'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:98:in `trigger_responding_cops'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:69:in `on_if'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:154:in `on_def'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:71:in `on_def'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:20:in `walk'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:86:in `investigate'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/team.rb:154:in `investigate_partial'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/team.rb:82:in `investigate'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:315:in `inspect_file'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:259:in `block in do_inspection_loop'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:293:in `block in iterate_until_no_changes'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:286:in `loop'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:286:in `iterate_until_no_changes'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:255:in `do_inspection_loop'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:138:in `block in file_offenses'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:163:in `file_offense_cache'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:137:in `file_offenses'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:128:in `process_file'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:109:in `block in each_inspected_file'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:108:in `each'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:108:in `reduce'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:108:in `each_inspected_file'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:94:in `inspect_files'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:47:in `run'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli/command.rb:11:in `run'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli/environment.rb:18:in `run'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli.rb:72:in `run_command'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli.rb:79:in `execute_runners'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli.rb:48:in `run'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/exe/rubocop:19:in `block in <top (required)>'
~/.rbenv/versions/3.1.3/lib/ruby/3.1.0/benchmark.rb:311:in `realtime'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/exe/rubocop:19:in `<top (required)>'
~/.rbenv/versions/3.1.3/bin/rubocop:25:in `load'
~/.rbenv/versions/3.1.3/bin/rubocop:25:in `<top (required)>'
~/.rbenv/versions/3.1.3/lib/ruby/3.1.0/bundler/cli/exec.rb:58:in `load'
~/.rbenv/versions/3.1.3/lib/ruby/3.1.0/bundler/cli/exec.rb:58:in `kernel_load'
~/.rbenv/versions/3.1.3/lib/ruby/3.1.0/bundler/cli/exec.rb:23:in `run'
~/.rbenv/versions/3.1.3/lib/ruby/3.1.0/bundler/cli.rb:486:in `exec'
~/.rbenv/versions/3.1.3/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
~/.rbenv/versions/3.1.3/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
~/.rbenv/versions/3.1.3/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
~/.rbenv/versions/3.1.3/lib/ruby/3.1.0/bundler/cli.rb:31:in `dispatch'
~/.rbenv/versions/3.1.3/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
~/.rbenv/versions/3.1.3/lib/ruby/3.1.0/bundler/cli.rb:25:in `start'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/libexec/bundle:48:in `block in <top (required)>'
~/.rbenv/versions/3.1.3/lib/ruby/3.1.0/bundler/friendly_errors.rb:120:in `with_friendly_errors'
~/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/libexec/bundle:36:in `<top (required)>'
~/.rbenv/versions/3.1.3/bin/bundle:25:in `load'
~/.rbenv/versions/3.1.3/bin/bundle:25:in `<main>'
.

1 file inspected, no offenses detected
Finished in 0.10952665500008152 seconds
```

## RuboCop version

```
$ bundle exec rubocop -V
1.40.0 (using Parser 3.1.3.0, rubocop-ast 1.24.0, running on ruby 3.1.3) [x86_64-linux]
```

**Comment by koic** [comment]
This issue was fixed by #11262. It will ship in the next release.
